mlir/python/mlir/_mlir_libs/__init__.py

```diff
@@ -56,14 +56,24 @@ def get_include_dirs() -> Sequence[str]:
 #
 # This facility allows downstreams to customize Context creation to their
 # needs.
+
+
+def get_registry():
+    if not hasattr(get_registry, "__registry"):
```

Comment:
Its' true but that's only for class fields - here I'm doing something even "dirtier" and setting it on the function object so it doesn't get mangled. 

But just this morning was thinking I'd refactor this to be a module global behind `threading.local()` instead of this hackery. Thoughts?

Reply:
Yeah, just move it up a level and use nonlocal. No need for dirty tricks that someone will need to grok later. I'm not sure it needs to be thread local. This is a really basic facility in the same vein as site_initialize, which is global-global.

Also, I note that your branch is named "remove_site_initialize_2". I assume this is all in addition to the current approach, which is used and works just fine for what we need it for.